### ISSUE TITLE
Screen Redraw Is Scoped To Initial Window Resolution

### ISSUE THREAD

**Issue by nbgibson** [hrkfdn/ncspot#740]
**Describe the bug**
This bug is partially in reference to #609. 'Ctrl + L' does well to resolve the "garbled text" issue seen when moving or resizing a window, but it redraws to the initial resolution of the terminal window when the program was started. Ideally the redraw call would poll for the existing window's dimensions and implement those rather than the cached value from initialization. This is observable in the current Windows build utilizing Powershell by way of the Windows Terminal. (I've also encountered similar issues at the time of launch of the program, but have been unable to reliably reproduce thus far.)

**To Reproduce**
1. Launch ncspot in the terminal emulator of your choice
2. Resize the emulator to a different size
3. Trigger the redraw command via 'Ctrl + L' hotkey command
4. If the new size is smaller than the initial window resolution, there were be "overdraw" of ncspot. Should the new resolution be larger, ncspot will take up a portion up to but not beyond the initial resolution (anchored to the top left corner).

**Expected behavior**
ncspot should resize to the current window's resolution at the time of the redraw command. Ideally this would be automatic, but one thing at a time.

**Screenshots**
Default:
![image](https://user-images.githubusercontent.com/10179709/152852782-a25bde3e-81a8-41e2-b79f-843923afc174.png)

Redrawing to a larger size:
![image](https://user-images.githubusercontent.com/10179709/152852381-5c1ef75a-b7c5-4fe9-b9c8-dcba119a5619.png)

Redrawing to a smaller size:
![image](https://user-images.githubusercontent.com/10179709/152852561-0a346064-65f5-4b3c-9435-061a4a58da24.png)

**System (please complete the following information):**
 - OS: Win10
 - Terminal: Windows Terminal (current)
 - Version: 0.9.5
 - Installed from: GitHub Release Page

**Backtrace/Debug log**
Can provide if needed.

**Additional context**
Comments on my music tastes are not solicited, but also not unwelcome. :v

**Comment by rashil2000** [comment]
I can also reproduce this issue (has been present since a long time now).

**Comment by hrkfdn** [comment]
Hmm, it could be a problem with pancurses/pdcurses not updating the screen size when the terminal is resized on Windows :thinking:

**Comment by rashil2000** [comment]
Yeah, I think so. There was an issue upstream https://github.com/gyscos/cursive/issues/461, but it seems to be related to Linux.

**Comment by rashil2000** [comment]
Would it be possible to manually fetch the terminal dimensions when a user explicitly calls the `resize` event (i.e. through pressing <kbd>Ctrl</kbd>+<kbd>L</kbd>?

**Comment by hrkfdn** [comment]
Not easily, there is also no option to redraw for a specific screen size. I believe this needs to be fixed upstream.

**Comment by Bettehem** [comment]
If I recall, building on Windows using `cursive/crossterm-backend` feature flag instead of `cursive/pancurses-backend` would allow the window to be resized.

**Comment by rashil2000** [comment]
The `cursive/crossterm-backend` feature supports more stuff like mouse scrolling and bold/italic text, but the huge disadvantage is the flicker - the entire screen redraws on each tiny change, making it unusable. Any way around that?

**Comment by Bettehem** [comment]
Not sure, I don't use Windows so I can't really help with that, sorry :/

**Comment by bogdan-calapod** [comment]
Jumping in to ask something that may be related - I also see that emoji/unicode characters are not drawn correctly in PowerShell (or in WSL via `ncspot.exe`):
![image](https://github.com/hrkfdn/ncspot/assets/5462727/49342a99-d813-4cda-acc9-5d9145f60a0c)

Although they are drawn correctly when installed in WSL and running the Ubuntu executable:

![image](https://github.com/hrkfdn/ncspot/assets/5462727/9c10e0ea-fc0a-4135-92f3-ca4e538822ce)

Should I open a new issue for this, or is it related ? Both screenshots are taken in Windows Terminal on Windows 11 22H2 (22621.2215)
